docs/api.md

```diff
@@ -0,0 +1,248 @@
+# API documentation
+
+- [Nomenclature](#nomenclature)
+- [Requirements for API consumers](#requirements-for-api-consumers)
+  - [Pass the Session Identifier to Access Restricted endpoints](#pass-the-session-identifier-to-access-restricted-endpoints)
+  - [Update the user's session if a new Session Identifier is returned](#update-the-users-session-if-a-new-session-identifier-is-returned)
+  - [End the user's session if an Access Restricted endpoint returns a `401: Unauthenticated`](#end-the-users-session-if-an-access-restricted-endpoint-returns-a-401-unauthenticated)
+  - [Ensure responses are not cached](#ensure-responses-are-not-cached)
+- [API endpoints](#api-endpoints)
+  - [`GET /api/oauth2/sign-in`](#get-apioauth2sign-in)
+  - [`POST /api/oauth2/callback`](#post-apioauth2callback)
+  - [`POST /api/oauth2/state`](#post-apioauth2state)
+  - [`GET /api/attributes`](#get-apiattributes)
+  - [`PATCH /api/attributes`](#patch-apiattributes)
+  - [`GET /api/transition-checker-email-subscription`](#get-apitransition-checker-email-subscription)
+  - [`POST /api/transition-checker-email-subscription`](#post-apitransition-checker-email-subscription)
+- [Healthcheck](#healthcheck)
+
+## Nomenclature
+
+- **Access Restricted**
+  - An API endpoint which requires a logged-in user.
+- **Identity Provider**
+  - The service which authenticates a user.  Currently we use the
+    [GOV.UK account manager prototype][].
+- **Session Identifier**
+  - An opaque token which identifies a user and provides access to
+    their attributes.
+
+[GOV.UK account manager prototype]: https://github.com/alphagov/govuk-account-manager-prototype/
+
+
+## Requirements for API consumers
+
+User sessions are managed in frontend apps using two custom headers:
+
+**Request:** ([see VCL](https://github.com/alphagov/govuk-cdn-config/blob/2ade45759be947a28b1225aee085311430bcbecf/vcl_templates/www.vcl.erb#L339-L340))
+
+- `GOVUK-Account-Session`: set by our CDN to the value of the user's session cookie.  This is the user's Session Identifier.
+
+**Response:** ([see VCL](https://github.com/alphagov/govuk-cdn-config/blob/2ade45759be947a28b1225aee085311430bcbecf/vcl_templates/www.vcl.erb#L442-L455))
+
+- `GOVUK-Account-Session`: sent to our CDN to update the user's session cookie.
+- `GOVUK-Account-End-Session`: sent to our CDN to delete the user's session cookie.
+
+Every app which uses this API is responsible for using the custom
+headers and implementing the behaviour in this section correctly.  We
+may provide a gem implementing this logic in the future.
+
+### Pass the Session Identifier to Access Restricted endpoints
+
+Some of these endpoints are Access Restricted, and require the
+`GOVUK-Account-Session` request header to be set.
+
+The [gds-api-adapters][] methods for these endpoints have a required
+`govuk_account_session:` parameter, which must be set to the latest
+Session Identifier.
+
```

Comment:
It looks to me like I, the developer on the service team who wants to integrate this service with the API, need to do two things here:

- set the `GOVUK-Account-Session` request header
- set the required `govuk_account_session:` parameter to the latest session identifier

Is that correct? Does setting one set the other?

Also, how do I set them? What do I actually need to do, and does this documentation need to tell me how to do this? @huwd suggested putting in an example request here.